crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs

```diff
@@ -364,23 +412,28 @@ fn fix_by_removing_imports<'a>(
     )
 }
 
-/// Generate a [`Fix`] to make bindings in a statement explicit, by changing from `import a` to
-/// `import a as a`.
-fn fix_by_reexporting<'a>(
+/// Generate a [`Fix`] to make bindings in a statement explicit, either by adding them to `__all__`
+/// or changing them from `import a` to `import a as a`.
+fn fix_by_reexporting(
     checker: &Checker,
     node_id: NodeId,
-    imports: impl Iterator<Item = &'a ImportBinding<'a>>,
+    imports: &[&ImportBinding],
+    dunder_all: Option<&ast::ExprList>,
 ) -> Result<Fix> {
     let statement = checker.semantic().statement(node_id);
-
-    let member_names = imports
-        .map(|binding| binding.import.member_name())
-        .collect::<Vec<_>>();
-    if member_names.is_empty() {
+    if imports.is_empty() {
         bail!("Expected import bindings");
     }
 
-    let edits = fix::edits::make_redundant_alias(member_names.iter().map(AsRef::as_ref), statement);
+    let edits = if let Some(dunder_all) = dunder_all {
+        fix::edits::add_to_dunder_all(imports.iter().map(|b| b.name), dunder_all)
+    } else {
+        let member_names = imports
+            .iter()
+            .map(|b| b.import.member_name())
+            .collect::<Vec<_>>();
```

Comment:
Is the `collect` here necessary, considering that you pass the iterator to `make_redundant_alias`?

Reply:
I'd like to avoid the collect if possible. I wasn't able to figure out how to obtain a reference to the result of calling `b.import.member_name()` on each element of the `imports` without collecting an intermediate vector.

Passing `imports.iter().map(|b| b.import.member_name())` directly into `make_redundant_alias` gets:
```
436 |         fix::edits::make_redundant_alias(imports.iter().map(|b| b.import.member_name()), statement)
    |         -------------------------------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `&str`, found `Cow<'_, str>`
```
So the closure passed to map must convert the `Cow<'_, str>` to `&str`, but using `.as_ref()` in that closure causes an ownership error:
```
437 |             imports.iter().map(|b| b.import.member_name().as_ref()),
    |                                    ----------------------^^^^^^^^^
    |                                    |
    |                                    returns a value referencing data owned by the current function
    |                                    temporary value created here
```
Suggestions are appreciated

Reply:
Hmm. Considering that `make_redundant_alias` is only called from `unused_import`. How about you move the implementation and simply make it accept a `Cow`? It doesn't seem worth figuring this puzzle out if we can make our lives easier by just changing the function signature ;)

Reply:
:shrug: ok!